python/src/uagents/asgi.py

```diff
@@ -149,6 +151,26 @@ async def handle_missing_content_type(self, headers: CaseInsensitiveDict, send):
                 }
             )
 
+    async def handle_get_messages(self, send):
+        """
+        Handle retrieval of stored messages.
+        """
+        messages = dispatcher.received_messages + self._dispenser.sent_messages
+        messages.sort(key=lambda x: x["timestamp"])
+        response = {"messages": [msg for msg in messages]}
```

Comment:
I think we can safely send the whole messages list to the UI for the frontend to filter, e.g. based on a timespan.